var/spack/repos/builtin/packages/py-datalad/package.py

```diff
@@ -33,38 +35,42 @@ class PyDatalad(PythonPackage):
     variant("duecredit", default=False, description="Enable duecredit support")
     variant("full", default=False, description="Enable support for all available variants")
 
+    depends_on("python@3.7:", when="@0.16:", type=("build", "run"))
     depends_on("python@3.6:", type=("build", "run"))
     depends_on("py-setuptools@40.8.0:", type="build")
 
     depends_on("git", type=("build", "run"))
     depends_on("git-annex", type=("build", "run"))
 
     # core
-    depends_on("py-appdirs", type=("build", "run"))
-    depends_on("py-chardet@3.0.4:", type=("build", "run"))
-    depends_on("py-distro", type=("build", "run"), when="^python@3.8:")
-    depends_on("py-importlib-metadata", type=("build", "run"), when="^python@:3.7")
+    depends_on("py-platformdirs", when="@0.16:", type=("build", "run"))
+    depends_on("py-chardet@3.0.4:4", type=("build", "run"))
+    depends_on("py-distro", when="^python@3.8:", type=("build", "run"))
+    depends_on("py-importlib-metadata@3.6:", when="^python@:3.10", type=("build", "run"))
+    depends_on("py-importlib-metadata", when="^python@:3.7", type=("build", "run"))
     depends_on("py-iso8601", type=("build", "run"))
     depends_on("py-humanize", type=("build", "run"))
     depends_on("py-fasteners@0.14:", type=("build", "run"))
     depends_on("py-packaging", type=("build", "run"), when="@0.15.4:")
     depends_on("py-patool@1.7:", type=("build", "run"))
     depends_on("py-tqdm", type=("build", "run"))
-    depends_on("py-wrapt", type=("build", "run"))
     depends_on("py-annexremote", type=("build", "run"))
+    depends_on("py-appdirs", when="@:0.15", type=("build", "run"))
+    depends_on("py-wrapt", when="@:0.15", type=("build", "run"))
     # additional dependency on 'colorama' if platform_system=="Windows"'
```

Comment:
Spack has Windows support now, can you add:
```python
depends_on("py-colorama", when="platform=windows", type=("build", "run"))
```

Reply:
Done